pkg/util/azureclient/authz/remotepdp/client.go

```diff
@@ -0,0 +1,90 @@
+package remotepdp
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// RemotePDPClient represents the Microsoft Remote PDP API Spec
+type RemotePDPClient interface {
+	CheckAccess(context.Context, AuthorizationRequest) (*AuthorizationDecisionResponse, error)
+}
+
+// remotePDPClient implements RemotePDPClient
+type remotePDPClient struct {
+	endpoint string
+	pipeline runtime.Pipeline
+}
+
+// NewRemotePDPClient returns an implementation of RemotePDPClient
+// endpoint - the fqdn of the regional specific endpoint of PDP
+// scope - the oauth scope required by the PDP server
+// cred - the credential of the client to call the PDP server
+func NewRemotePDPClient(endpoint, scope string, cred azcore.TokenCredential) RemotePDPClient {
```

Comment:
is there a specific reason why we are returning the interface type there? returning the actual type would make more sense otherwise and you'd lose no functionality, plus you'd be able to test that function without needing type casting. 

Plus having some testing on public functions is always nice

Reply:
*misclicked and accidentally deleted my previous comment...

I was trying to be consistent with how the same thing is done in the ARO-RP code base.  Maybe I've mistaken it for a pattern, but you can look at these functions as they return an interface type:
- refreshable.NewAuthorizer
- validate.NewOpenshiftClusterDynamicValidator
- instancemetadata.New
- dynamic.NewValidator

I don't mind making it return the concrete type though, just let me know what the right thing to do.

For testing, I couldn't find a good example of how to do this.  Do you have any suggestions?  I was thinking about using mocks but I'm not sure what tests should actually be performed on this data compositional logic.

Reply:
yeah I know we are using that pattern, but it never was a good pattern. the problem with this pattern is that you lose the type, and so you cannot access the fields anymore unless you do some type casting but this is very cumbersome and error prone. If you return the concrete type, it will be much easier to test. 

Are any of the fields relying on distant services or is everything local ? I'd just test that we are creating the objects with the proper fields

Reply:
Everything is local until the client is used.  I have a created a test to call this function with the expected value.  Let me know if this is ok.